core/src/main/scala/org/genivi/sota/core/campaigns/CampaignLauncher.scala

```diff
@@ -9,16 +9,27 @@ import cats.implicits._
 import org.genivi.sota.common.DeviceRegistry
 import org.genivi.sota.core.UpdateService
 import org.genivi.sota.core.data.{Campaign, UpdateRequest}
-import org.genivi.sota.core.db.Campaigns
+import org.genivi.sota.core.db.{Campaigns, UpdateSpecs}
 import org.genivi.sota.data.{Interval, Namespace, PackageId, Uuid}
-import slick.driver.MySQLDriver.api.Database
+import slick.driver.MySQLDriver.api._
 
 import scala.concurrent.{ExecutionContext, Future}
 
 object CampaignLauncher {
   import Campaign._
   import UpdateService._
 
+  def cancel(id: Campaign.Id)(implicit db: Database, ec: ExecutionContext): Future[Unit] = {
+    val dbIO = Campaigns.fetchGroups(id).flatMap { grps =>
+      DBIO.seq(grps.collect {
+                 case CampaignGroup(_, Some(ur)) =>
+                   UpdateSpecs.cancelAllUpdatesByRequest(ur)
+               } :_*)
+    }
+
+    db.run(dbIO.transactionally)
+  }
+
```

Comment:
Think this would be a bit more readable (?)

```scala
    val dbIO = Campaigns.fetchGroups(id).flatMap { grps =>
      val cancelIO = grps.collect {
        case CampaignGroup(_, Some(ur)) =>
          UpdateSpecs.cancelAllUpdatesByRequest(ur)
      }

      DBIO.sequence(cancelIO).map(_ => ())
    }
    
    db.run(dbIO.transactionally)
```

Or if you want to use `seq`, `DBIO.sequence(cancelIO: _*)` but I don't like using the exploding mouth face.